components/net/http_loader.rs

```diff
@@ -1588,12 +1569,28 @@ async fn http_network_or_cache_fetch(
 
         // Step 14.3 If request’s use-URL-credentials flag is unset or isAuthenticationFetch is true, then:
         if !http_request.use_url_credentials || authentication_fetch_flag {
-            // TODO(#33616, #27439): Prompt the user for username and password from the window
+            let Some(credentials) = prompt_user_for_credentials(&context.state.embedder_proxy)
```

Comment:
so I've updated this part to follow step 14.4 https://fetch.spec.whatwg.org/#ref-for-proxy-authentication-entry%E2%91%A0. Gotta scroll up a bit to see it. 

It's working so far. Let me know if I'm missing anything @jdm